lib/guest_installaltion_and_configuration_base.pm

```diff
@@ -0,0 +1,1428 @@
+        record_info("Detached $self->{guest_name} installation screen process $self->{guest_installation_session} successfully", "Well Done !");
+        $self->get_guest_installation_session if ($self->{guest_installation_session} eq '');
+        type_string("reset\n");
+        wait_still_screen;
+    }
+    else {
+        record_info("Failed to detach $self->{guest_name} installation screen process $self->{guest_installation_session}", "Bad luck !");
+        reset_consoles;
+        select_console('root-ssh');
+        $self->get_guest_installation_session if ($self->{guest_installation_session} eq '');
+        type_string("reset\n");
+        wait_still_screen;
+    }
+    $self->{guest_installation_attached} = 'false';
+    return $self;
+}
+
+#Record final guest installation result in [guest_installation_result] and set [stop_run] and [stop_timestamp].
+sub record_guest_installation_result {
+    my ($self, $_guest_installation_result) = @_;
+    $_guest_installation_result //= '';
+
+    $self->reveal_myself;
+    $self->{guest_installation_result} = $_guest_installation_result;
+    record_info("Guest $self->{guest_name} installation has already been marked as $self->{guest_installation_result}", "It is done !");
+    $self->{stop_run}       = time();
+    $self->{stop_timestamp} = localtime($self->{stop_run});
+    return $self;
+}
+
+#Collect guest y2logs via ssh and save guest config xml file.
+sub collect_guest_installation_logs_via_ssh {
+    my $self = shift;
+
+    $self->reveal_myself;
+    $self->get_guest_ipaddr;
+    if ((script_run("nmap $self->{guest_ipaddr} -PN -p ssh | grep -i open") eq 0) and ($self->{guest_ipaddr} ne '') and ($self->{guest_ipaddr} ne 'NO_IP_ADDRESS_FOUND_AT_THE_MOMENT')) {
+        record_info("Guest $self->{guest_name} can be reached by ssh via ip address $self->{guest_ipaddr}.", "Going to collect logs via ssh");
+        script_run("ssh -vvv -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root\@$self->{guest_ipaddr} \"save_y2logs /tmp/$self->{guest_name}_y2logs.tar.gz\"");
+        script_run("scp -r -vvv -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root\@$self->{guest_ipaddr}:/tmp/$self->{guest_name}_y2logs.tar.gz $self->{guest_log_folder}");
+    }
+    else {
+        record_info("Guest $self->{guest_name} has no ssh connection available.Not able to collect logs from it via ssh", "Guest ip address is $self->{guest_ipaddr}");
+    }
+    script_run("virsh dumpxml $self->{guest_name} > $self->{guest_log_folder}/virsh_dumpxml_$self->{guest_name}.xml");
+    script_run("rm -f -r $common_log_folder/autoyast_$self->{guest_name}*");
+    return $self;
+}
+
+#Upload logs collect by collect_guest_installation_logs_via_ssh.
+sub upload_guest_installation_logs {
+    my $self = shift;
+
+    $self->reveal_myself;
+    assert_script_run("tar czvf /tmp/guest_installation_logs.tar.gz $common_log_folder");
+    upload_logs("/tmp/guest_installation_logs.tar.gz");
```

Comment:
Verification run http://10.67.129.106/tests/1103 already provided virt-install log. Now guest_installation_and_configuration_base.pm has virt_utils::collect_host_and_guest_logs in post_fail_hook

Reply:
Glad to see that. Thanks for the improvement.

Regarding logs, I checked the log for verfication job http://10.67.129.106/tests/1086#( Multiple kvm guests, one of them failed). Some  improvement can be done:
- duplicated log in virt_logs_all and guest_installation_log, eg nmap_subnets_scan_results
- duplicated log in guest_installation_log itself, eg autoyast file per guest is uploaded to both  guest_name subdir and log root dir for successful guests
- multiple sle-15-sp3-64-fv-uefi-3_installation_log in sle-15-sp3-64-fv-uefi-3 subdir: what's the difference of these logs? can it be reflected in the log name?

The screenshot for the two logs for easier view:

Reply:
![Screenshot_20210427_102627-guest-installation-log](https://user-images.githubusercontent.com/5440789/116182727-4be9ed00-a70c-11eb-8f4b-62d63d630f87.png)
![Screenshot_20210427_102657-virt-logs-all](https://user-images.githubusercontent.com/5440789/116182732-4db3b080-a70c-11eb-8115-d8a819442354.png)

Reply:
Item1: They are collected at different time and tagged with timestamp. So they can be all kept.
Item2: It is outdated. It is collected from old test run
Item3:Because guest is disconnected somehow, when they are re-connected a install log with a new timestamp will be generated. This is reasonable.